plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -490,7 +491,12 @@ public synchronized void dropDatabase(ConnectorSession session, String schemaNam
                 }
             }).orElse(deleteSchemaLocationsFallback);
 
-            delegate.dropDatabase(schemaName, deleteData);
+            if (cascade) {
```

Comment:
If this statement fails half-through we'll not be able to recover the dropped tables.
Is this acceptable?

Reply:
That was the main concern I was preparing this PR. I will look into if we can improve the behavior.

Reply:
I don't think we can implement an all-or-nothing DROP SCHEMA CASCADE in Hive connector. Which metastores allow to transactionally delete multiple tables and then schema?

At best, we can try to detect foreseeable problems (for example any non-Hive tables present? any Iceberg MV entries that hive connector doesn't know how to drop?)